backend/src/layer/logic/mealplan_management/test/meal_plan_parser_mock.rs

```diff
@@ -0,0 +1,85 @@
+//! This crate contains mocks of [`MealPlanParserMock`] for testing.
+
+use async_trait::async_trait;
+
+use crate::interface::mensa_parser::ParseError;
+use crate::{
+    interface::mensa_parser::{
+        model::{Dish, ParseCanteen, ParseLine},
+        MealplanParser,
+    },
+    util::{
+        Additive::Alcohol,
+        Allergen::{Er, Pe},
+        Date,
+        MealType::Vegan,
+        Price,
+    },
+};
+
+fn get_dish(dish_number: u32) -> Dish {
+    Dish {
+        name: format!("Autogenerated Dish number {dish_number}"),
+        price: Price {
+            price_student: 1000,
+            price_employee: 1,
+            price_guest: 2,
+            price_pupil: 3,
+        },
+        allergens: vec![Pe, Er],
+        additives: vec![Alcohol],
+        meal_type: Vegan,
+    }
+}
+
+fn get_dishes(dish_amount: u32) -> Vec<Dish> {
+    let mut dishes = Vec::new();
+    for i in 1..dish_amount {
+        dishes.push(get_dish(i));
+    }
+    dishes
+}
+
+fn get_line(line_number: u32, dish_amount: u32) -> ParseLine {
+    ParseLine {
+        name: format!("Autogenerated Line number {line_number}"),
+        dishes: get_dishes(dish_amount),
+    }
+}
+
+fn get_lines(line_amount: u32, dish_amount: u32) -> Vec<ParseLine> {
+    let mut lines = Vec::new();
+    for i in 1..line_amount {
+        lines.push(get_line(i, dish_amount));
+    }
+    lines
+}
+
+fn get_canteen(canteen_number: u32, line_amount: u32, dish_amount: u32) -> ParseCanteen {
+    ParseCanteen {
+        name: format!("Autogenerated Canteen number {canteen_number}"),
+        lines: get_lines(line_amount, dish_amount),
+    }
+}
+
+fn get_canteens(canteen_amount: u32, line_amount: u32, dish_amount: u32) -> Vec<ParseCanteen> {
+    let mut canteens = Vec::new();
+    for i in 1..canteen_amount {
+        canteens.push(get_canteen(i, line_amount, dish_amount));
+    }
+    canteens
+}
+
+pub struct MealPlanParserMock;
+
+#[async_trait]
+impl MealplanParser for MealPlanParserMock {
+    /// Initiate a parse procedure. Returns a canteen struct containing mealplan data of the given date.
```

Comment:
no duplicate copy of doc comment to trait definition

Reply:
The trait function descriptions should be equal to the implementations?

Reply:
if you remove them, they will still be shown. But it is not that important here as these are only mocks